Microsoft.Azure.CosmosRepository/src/Providers/DefaultCosmosContainerDefaultTimeToLiveProvider.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) IEvangelist. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using Microsoft.Azure.CosmosRepository.Builders;
+using Microsoft.Azure.CosmosRepository.Options;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.CosmosRepository.Providers
+{
+    /// <inheritdoc/>
+    class DefaultCosmosContainerDefaultTimeToLiveProvider : ICosmosContainerDefaultTimeToLiveProvider
+    {
+        private readonly IOptions<RepositoryOptions> _options;
+
+        /// <summary>
+        /// Creates an instance of the <see cref="DefaultCosmosContainerDefaultTimeToLiveProvider"/>.
+        /// </summary>
+        /// <param name="options">The repository options.</param>
+        public DefaultCosmosContainerDefaultTimeToLiveProvider(IOptions<RepositoryOptions> options) => _options = options;
+
+        /// <inheritdoc/>
+        public int GetDefaultTimeToLive<TItem>() where TItem : IItem
+        {
+            ContainerOptionsBuilder options = _options.Value.GetContainerOptions<TItem>();
+
+            if (options is null)
+            {
+                return -1;
+            }
+
+            foreach (ContainerOptionsBuilder containerOptions in _options.Value.ContainerOptions.Where(co => co.Name == options.Name))
+            {
+                if (containerOptions.ContainerDefaultTimeToLive != null && containerOptions.ContainerDefaultTimeToLive != options.ContainerDefaultTimeToLive)
+                {
+                    throw new InvalidOperationException($"The container {options.Name} has conflicting default time to live values. " +
```

Comment:
Maybe something we could do in the future to evolve this library a bit more is document and promote official error codes and messages in a look up for consumers? With troubleshooting help for common difficulties?

Reply:
Yes I agree, how would you do that a custom exception from our library?

Reply:
We could certainly start by creating exceptions, but I was thinking we could either build out a Wiki here too or even a stand alone (but that's a bigger effort). Or maybe not, I could probably spin up a Hugo doc site in an afternoon.

Reply:
Yeah that would be really cool, could that be hosted on Github pages?

Reply:
Indeed, that could.

Reply:
That would be great, is it markdown based?

Reply:
Yes, entirely markdown based. I'll see what I can do, but time is something I don't have a lot of lately. :)

Reply:
I can always take a look if we get the repo up and working?